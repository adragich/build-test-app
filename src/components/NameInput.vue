<template>
  <div class="build-test-u-flex-column build-test-u-spacer">
    <label class="build-test__input-label build-test-u-spacer build-test-u-lightgrey-color">
      Kindly asking you to provide your name
    </label>
    <input
      v-model="innerValue"
      type="text"
      class="build-test__input build-test-u-spacer"
      placeholder="Jhon Doe ðŸ˜Ž"
      @input="$emit('input', $event.target.value)"
    >
  </div>
</template>

<script>
export default {
  name: 'NameInput',
  props: {
    name: {
      type: String,
      required: true
    }
  },
  data: function () {
    return {
      innerValue: this.name,
    };
  },
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
  .build-test__input {
    border: 1px solid #dfdfdf;
    line-height: 32px;
    width: 280px;
    max-width: 100%;
    border-radius: 8px;
    box-shadow: 0 2px 1px;
    padding: 0 16px;
    font-size: 16px;
    font-family: inherit;
  }
</style>
