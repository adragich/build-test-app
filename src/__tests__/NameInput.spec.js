import { shallowMount } from '@vue/test-utils';
import NameInput from '../components/NameInput.vue';

describe('NameInput.vue', () => {
  test('renders name', () => {
    const name = 'Anastasia';
    const wrapper = shallowMount(NameInput, {
        propsData: { name }
    });
    const textInput = wrapper.find('input');
    
    expect(textInput.element.value).toBe(name);
  });

  test('emitting "input" event on input', () => {
    const wrapper = shallowMount(NameInput, {
        propsData: { 'name': '' }
    });
    let textInput = wrapper.find('input');

    textInput.trigger('input', 'John');
    
    expect(wrapper.emitted('input')).toBeTruthy();
  });

  test('renders textarea item correctly', () => {
    const wrapper = shallowMount(NameInput, {
        propsData: { 'name': '' }
    });
    
    expect(wrapper.element).toMatchSnapshot();
  });
});
