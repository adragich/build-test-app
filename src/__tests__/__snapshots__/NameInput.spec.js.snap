// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameInput.vue renders textarea item correctly 1`] = `
<div
  class="build-test-u-flex-column build-test-u-spacer"
>
  <label
    class="build-test__input-label build-test-u-spacer build-test-u-lightgrey-color"
  >
    
    Kindly asking you to provide your name
  
  </label>
   
  <input
    class="build-test__input build-test-u-spacer"
    placeholder="Jhon Doe ðŸ˜Ž"
    type="text"
  />
</div>
`;
